### PR TITLE
Run tests on CockroachDB

### LINKED ISSUE #98: Test with CockroachDB
We should run our tests (both locally and on CI) with CockroachDB. There is no separate handler for it; instead, a regular `pg` handler should be used. We start CockroachDB side-by-side with PostgreSQL (and Tigris), don't try to replace the latter with the former.

Our changes in Taskfile(s) and CI configuration should be more or less copy&pasted. We might want to do small refactoring, but that's not required. We will revisit that at #85.

But changes in the `envtool`, if any, should be good; its refactoring is in scope.

DoD:
* tests could be run agains CockroachDB both locally (with `task`) and on CI;
* it should be possible to run FerretDB with CockroachDB with `task run-cockroach`.
* if we see that CockoachDB is not fully compatible, let's document what doesn't work.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

This PR closes #98.

* [x] run tests against CockroachDB locally (with task)
* [x] run tests against CockroachDB on CI;
* [x] run FerretDB with CockroachDB with `task run-cockroach`.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ tasks:
       - >
         docker-compose up --always-recreate-deps --force-recreate --remove-orphans --renew-anon-volumes --timeout=0 --detach
         --build
-        postgres tigris mongodb
+        postgres tigris cockroach mongodb
```
Comment (AlekSi) [inline Taskfile.yml:81]:
```suggestion
        postgres tigris cockroach mongodb
```

File: build/deps/cockroach.Dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+FROM cockroachdb/cockroach:latest
```
Comment (AlekSi) [inline build/deps/cockroach.Dockerfile:1]: Let's use a tagged release. Dependabot would update it

File: integration/setup/common.go [hunk 3]
```diff
@@ -44,6 +44,10 @@ var (
 	handlerF    = flag.String("handler", "pg", "handler to use for in-process FerretDB")
 	compatPortF = flag.Int("compat-port", 37017, "second system's port for compatibility tests; if 0, they are skipped")
 
+	pgHostF     = flag.String("pg-host", "127.0.0.1:5432", "PostgreSQL database hostname")
+	pgUserF     = flag.String("pg-user", "postgres", "PostgreSQL database username")
+	pgDatabaseF = flag.String("pg-db", "ferretdb", "PostgreSQL database name")
```
Comment (AlekSi) [inline integration/setup/common.go:49]: Let's replace it with the full URL and use it for PostgreSQL too

File: docker-compose.yml [hunk 4]
```diff
@@ -29,6 +29,19 @@ services:
     environment:
       - TIGRIS_SERVER_LOG_LEVEL=info
 
+  cockroach:
+    build:
+      context: ./build/deps
+      dockerfile: cockroach.Dockerfile
+    container_name: ferretdb_cockroach
+    command: start-single-node --insecure
+
```
Comment (AlekSi) [inline docker-compose.yml:38]:
```suggestion
```

File: Taskfile.yml (hunk 1 above)
Comment (noisersup) [reply Taskfile.yml:81]: That's weird I've checked for trailing spaces

**Comment by noisersup** [comment]
I realised that tests in this PR weren't run on Cockroach. I can see rn that there are a lot of tests failing and there's probably a lot of work to do

File: Taskfile.yml [hunk 5]
```diff
@@ -147,6 +147,7 @@ tasks:
     deps:
       - test-integration-pg
       - test-integration-tigris
+      - test-integration-cockroach
```
Comment (AlekSi) [inline Taskfile.yml:150]: Let's not run them by default, but please update task's description ("all" is not valid in that case)

Comment (noisersup) [reply Taskfile.yml:150]: "Run all stable integration tests"?
"Run integration tests for PostgreSQL and Tigris"?

Comment (AlekSi) [reply Taskfile.yml]: I would mention names, yeah

File: internal/util/testutil/pg.go [hunk 6]
```diff
@@ -53,20 +53,28 @@ func PostgreSQLURL(tb testing.TB, opts *PostgreSQLURLOpts) string {
...
-	}
+	u.RawQuery = q.Encode()
 
 	return u.String()
 }
+
+// AddTestParams adds testing parameters to the query.
+func AddTestParams(uri *url.URL) {
+	q := uri.Query()
+	q.Set("pool_min_conns", "1")
```
Comment (AlekSi) [inline internal/util/testutil/pg.go:78]: Let's add to the `PostgreSQLURLF` itself and all `--postgresql-url` non-default values, and remove this function

**Review by rumyantseva — APPROVED** [review]
LGTM:
- Cockroach image is in place
- A new task in `Taskfile` is in place
- The changes look reasonable

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
(but Sprint field wasn't set)

**Pull request merged**